lib/src/components/NxCombobox/types.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import * as PropTypes from 'prop-types';
+import { omit } from 'ramda';
+import { HTMLAttributes, ReactNode } from 'react';
+import { PublicProps as NxTextInputProps, propTypes as nxTextInputPropTypes } from '../NxTextInput/types';
+
+import DataItem from '../../util/DataItem';
+
+type InputProps = Omit<NxTextInputProps, 'type' | 'value' | 'isPristine' | 'readOnly'>;
+
+export interface Props<T extends string | number = string> extends Omit<HTMLAttributes<HTMLDivElement>, 'onSelect'> {
+  searchText: string;
+  loading?: boolean | null;
+  error?: ReactNode;
```

Comment:
Since we're also going to need validation errors, let's change this to `loadError` to be less ambiguous

Reply:
Updated here cf6a0ff75fa273c48e5523cad172ba37b5f7fb95.